x-pack/plugins/maps/public/routing/routes/maps_app/load_map_and_render.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { MapsAppView } from '.';
+import { getMapsSavedObjectLoader } from '../../bootstrap/services/gis_map_saved_object_loader';
+import { getToasts } from '../../../kibana_services';
+import { i18n } from '@kbn/i18n';
+import { Redirect } from 'react-router-dom';
+
+export const LoadMapAndRender = class extends React.Component {
+  state = {
+    savedMap: null,
+    failedToLoad: false,
+  };
+
+  componentDidMount() {
+    this.loadSavedMap();
```

Comment:
order appears incorrect. It only works by accident due to async nature of the saved-object loading.

flip to be consistent with other react-component in Maps

```
this._isMounted = true; //flag component as ready
this._loadSavedMap(); //do the thing
```